cmd/scan/scan.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	_ "log"
+	"os"
+	"path/filepath"
+
+	"strconv"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcutil"
+)
+
+type Address struct {
+	Addr         string `json:"address"`
+	MinScanBlock int64  `json:"min_scan_block"`
+	MidScanBlock int64  `json:"mid_scan_block"`
+	MaxScanBlock int64  `json:"max_scan_block"`
+	Txs          []Tx   `json:"txs"`
+}
+
+type Tx struct {
+	TxHash      string `json:"tx_hash"`
+	BlockHash   string `json:"block_hash"`
+	ParentHash  string `json:"parent_hash"`
+	BlockHeight int64  `json:"block_height"`
+}
+
+type Deposit struct {
+	Addr string
+	Tx   Tx
+}
+
+func ScanBlock(client *rpcclient.Client, blockID int64) ([]Deposit, error) {
+	blockHash, err := client.GetBlockHash(blockID)
+	if err != nil {
+		return nil, err
+	}
+	block, err := client.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		return nil, err
+	}
+	parentHash := block.PreviousHash
+	var deposits []Deposit
+	var depTx Tx
+	depTx.BlockHeight = blockID
+	depTx.BlockHash = blockHash.String()
+	depTx.ParentHash = parentHash
+	for _, tx := range block.RawTx {
+		for i, addr := range tx.Vout {
+			depTx.TxHash = tx.Txid
+			depTx.TxHash = depTx.TxHash + ":" + strconv.Itoa(i)
+			if len(addr.ScriptPubKey.Addresses) > 0 {
+				deposits = append(deposits, Deposit{Addr: addr.ScriptPubKey.Addresses[0], Tx: depTx})
+			}
+		}
+
+	}
+	return deposits, nil
+}
+
+func CompareAddress(addr Address, deps []Deposit) Address {
+	for _, dep := range deps {
+		if addr.Addr == dep.Addr {
+			if addr.Txs[0].BlockHeight == -1 {
```

Comment:
I don't see any code that sets `BlockHeight` to `-1` in the first place

Reply:
I don't how addresses will add to wallet, for now it's just putting  empty address struct in wallet.json. That's why i use blockheight=-1 as flag for zero count txs

Reply:
This requires the user to know to put -1. Which will lead to an error.

The Txs array should be empty if there are no Txs. It should handle the initial/null case differently

Reply:
My way was wrong, really. I fix all and added options for adding addrs from file or from cmd.